src/cache.jl

```diff
@@ -94,7 +94,11 @@ end
 updatecache(absentmodule::Symbol, server) = updatecache([absentmodule], server)
 
 function updatecache(absentmodules::Vector{Symbol}, server)
+    server.cp_lock && return
+
+    server.cp_lock = true
     o,i, p = readandwrite(`$JULIA_HOME/julia -e "include(\"packages/LanguageServer/src/cache.jl\");
+    delete!(Base.ENV, \"JULIA_PKGDIR\");
```

Comment:
Hm, there might be a Windows/Unix difference here... The current theory is this: VS Code starts the language server and sets its ``JULIA_PKGDIR`` to a folder under the VS Code extension. That results in all the precompile files being there, separate from the normal user location.

At least on windows, the child process that we start here in code does not inherit that environmental variable setting (I'm 99% sure), instead the child process has ``JULIA_PKGDIR`` point towards the normal user package location, which is what we want.

I'll prep a new PR that makes sure our child processes always have the right value for ``JULIA_PKGDIR`` there, should be fairly simple. I don't think deleting ``JULIA_PKGDIR`` is a good idea, because then things like the location for precompile etc. are really left unclear. Instead, we should start the child process with the correct value for that env variable.